pkg/shell/cockpit-docker.js

```diff
@@ -773,29 +774,251 @@ PageRunImage.prototype = {
+                    /* if they are empty, make sure they are both cleared of errors */
+                    clear_control_error(input_ports[0], 0);
+                    clear_control_error(input_ports[1], 1);
+                }
+            });
+
+            /* check ports */
+            for (var port_index = 0; port_index < exposed_ports.container.length; ++port_index) {
+                check_port(exposed_ports.container, exposed_ports.protocol, port_index, 0);
+                check_port(exposed_ports.host, exposed_ports.protocol, port_index, 1);
+            }
+
+            /* update run status */
+            check_entries_valid();
+        }    
+
+        function highlight_command_error() {
+            $('#containers-run-image-command-note').removeClass('containers-run-message');
+            $('#containers-run-image-command').parent().addClass('has-error');
+        }
+
+        function hide_command_error() {
+            $('#containers-run-image-command-note').addClass('containers-run-message');
+            $('#containers-run-image-command').parent().removeClass('has-error');
+        }
+
+        function check_command() {
+            if ($('#containers-run-image-command').val() === "")
+                highlight_command_error();
+            else
+                hide_command_error();
+
+            /* update run status */
+            check_entries_valid();
+        }
+
+        function show_link_message(control, message_type, message, help_index) {
+            control.parent().addClass(message_type);
+            var help_item = help_item_for_control(control, help_index);
+            help_item.text(message);
+            var err_item = help_item.parent();
+            err_item.addClass(message_type);
+            err_item.removeClass('containers-run-message');
+        }
+
+        function check_alias(aliases, alias_index ) {
+            var alias = aliases[alias_index];
+            var alias_value = alias.val();
+
+            clear_control_error(alias, 1);
+
+            /* check for empty field */
+            if (alias_value === "") {
+                /* still valid if empty */
+                show_link_message(alias, 'has-error', _("No alias specified"), 1);
+                return;
+            }
+
+            /* check for duplicate entries */
+            for (var i = 0; i < aliases.length; ++i) {
+                if (i === alias_index)
+                    continue;
+                var second_alias = aliases[i];
+                if ((alias_value === second_alias.val())) {
+                    show_link_message(alias, 'has-error', _("Duplicate alias"), 1);
+                    return;
+                }
+            }
+        }
+
+        function check_links() {
+            var aliases = [];
+            var containers = [];
+            /* gather all aliases */
+            $('#select-linked-containers').children('form').each(function() {
+                var container = $($(this).find('select'));
```

Comment:
> Why the double jQuery wrapper? .find() returns a jQuery object just fine.

I remember having issues that some of the returned values weren't jQuery objects. I will test without double wrapper.